frameworks/js-bindings/bindings/manual/network/XMLHTTPRequest.cpp

```diff
@@ -724,10 +731,28 @@ JS_BINDED_FUNC_IMPL(MinXmlHttpRequest, send)
     _setHttpRequestHeader();
     _sendRequest(cx);
     _notify(_onloadstartCallback);
+    
+    //begin schedule for timeout
+    if(_timeout > 0)
+    {
+        _scheduler->scheduleUpdate(this, 0, false);
```

Comment:
Why not use `HttpClient::setTimeoutForConnect` & `HttpClient::setTimeoutForRead` which will set the real timeout parameter for connection and read? While using `scheduler` which is a timer of engine isn't suitable here.